src/couch_db_updater.erl

```diff
@@ -1454,3 +1450,13 @@ default_security_object(_DbName) ->
         "everyone" ->
             []
     end.
+
+delete_compact_files(RootDir, FullFilepath) ->
+    lists:foreach(fun(Ext) ->
+        couch_file:delete(RootDir, FullFilepath ++ Ext)
+    end, [".compact", ".compact.data", ".compact.meta"]).
+
+purge_compact_files(FullFilepath) ->
+    lists:foreach(fun(Ext) ->
+        file:delete(FullFilepath ++ Ext)
+    end, [".compact", ".compact.data", ".compact.meta"]).
```

Comment:
Since you repeat this list, any reason to not create a 
`define(COMPACT_SUFFIXES, [".compact", ".compact.data", ".compact.meta"]).`
?